File: packages/papaya/backend/sirius-components-papaya/src/main/java/org/eclipse/sirius/components/papaya/Project.java
/*******************************************************************************
 * Copyright (c) 2024, 2026 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/
package org.eclipse.sirius.components.papaya;

/**
 * <!-- begin-user-doc --> A representation of the model object '<em><b>Project</b></em>'. <!-- end-user-doc -->
 *
 * <p>
 * The following features are supported:
 * </p>
 * <ul>
 *   <li>{@link org.eclipse.sirius.components.papaya.Project#getHomepage <em>Homepage</em>}</li>
 * </ul>
 *
 * @see org.eclipse.sirius.components.papaya.PapayaPackage#getProject()
 * @model
 * @generated
 */
public interface Project extends NamedElement, Container {
    /**
	 * Returns the value of the '<em><b>Homepage</b></em>' attribute.
	 * <!-- begin-user-doc --> <!-- end-user-doc -->
	 * @return the value of the '<em>Homepage</em>' attribute.
	 * @see #setHomepage(String)
	 * @see org.eclipse.sirius.components.papaya.PapayaPackage#getProject_Homepage()
	 * @model
	 * @generated
	 */
    String getHomepage();

    /**
	 * Sets the value of the '{@link org.eclipse.sirius.components.papaya.Project#getHomepage <em>Homepage</em>}' attribute.
	 * <!-- begin-user-doc --> <!-- end-user-doc -->
	 * @param value the new value of the '<em>Homepage</em>' attribute.
	 * @see #getHomepage()
	 * @generated
	 */
    void setHomepage(String value);

} // Project

File: integration-tests-cypress/cypress/pages/Project.ts
/*******************************************************************************
 * Copyright (c) 2023, 2025 Obeo.
 * This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Obeo - initial API and implementation
 *******************************************************************************/

import { RenameProjectDialog } from './RenameProject';

export class Project {
  public visit(
    projectId: string,
    representationId?: string,
    options?: Partial<Cypress.VisitOptions>
  ): Cypress.Chainable<Cypress.AUTWindow> {
    return cy.visit(`/projects/${projectId}/edit${representationId ? '/' + representationId : ''}`, {
      ...options,
      onBeforeLoad(win) {
        cy.spy(win.console, 'debug').as('consoleDebug');
      },
    });
  }
  public disableDeletionConfirmationDialog(): void {
    cy.window().then((win) => {
      win.localStorage.setItem('sirius-confirmation-dialog-disabled', JSON.stringify(true));
    });
  }
  public enableDeletionConfirmationDialog(): void {
    cy.window().then((win) => {
      win.localStorage.setItem('sirius-confirmation-dialog-disabled', JSON.stringify(false));
    });
  }

  public rename(newProjectName: string): void {
    this.isLoaded();
    const renameDialog = this.getProjectNavigationBar().getRenameDialog();
    renameDialog.clearValue();
    renameDialog.getRenameTextField().type(newProjectName);
    renameDialog.validate(true);
    renameDialog.getRenameButton().click();
  }

  public getProjectNavigationBar(): ProjectNavigationBar {
    this.isLoaded();
    cy.getByTestId(`navbar-title`).should('be.visible');
    cy.getByTestId('navigation-bar').findByTestId('more').click();
    return new ProjectNavigationBar();
  }

  private isLoaded(): void {
    cy.get('@consoleDebug').should('be.calledWith', 'query getProjectAndRepresentation: response received');
    cy.get('@consoleDebug').should('be.calledWith', 'query getOnboardData: response received');
    cy.get('@consoleDebug').should('be.calledWith', 'query getRepresentationMetadata: response received');
    cy.get('@consoleDebug').should('be.calledWith', 'query getAllTreeFilters: response received');
    cy.get('@consoleDebug').should('be.calledWith', 'query getAllExplorerDescriptions: response received');
  }
}

class ProjectNavigationBar {
  public getDownloadLink(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.getByTestId('download-link');
  }

  public getRenameButton(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.getByTestId('rename');
  }

  public getRenameDialog(): RenameProjectDialog {
    this.getRenameButton().should('exist').click();
    return new RenameProjectDialog();
  }

  public getShareButton(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.getByTestId('share');
  }

  public getShareDialog(): ShareProjectDialog {
    this.getShareButton().should('exist').click();
    return new ShareProjectDialog();
  }

  public getDeleteButton(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.getByTestId('delete');
  }

  public getSettingsButton(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.getByTestId('project-settings-link');
  }
}

class ShareProjectDialog {
  public getSharePathTextField(): Cypress.Chainable<JQuery<HTMLElement>> {
    return cy.getByTestId('share-path');
  }
}
